controllers/searchcontroller/enterprise_search_source_test.go

```diff
@@ -215,35 +213,6 @@ func TestEnterpriseResourceSearchSource_Validate(t *testing.T) {
 			authModes:    []string{"SCRAM", "SCRAM-SHA-1", "SCRAM-SHA-256"},
 			expectError:  false,
 		},
-		// Internal cluster authentication tests
-		{
-			name:                "X509 internal cluster auth not supported",
```

Comment:
shouldn't we keep this test case and just verify that we don't throw an error? As I understand there are three options for `internalClusterAuth` right now:
 - SCRAM
 - X5909
 - empty

Shouldn't we validate if received `internalClusterAuth` that is outside of the accepted values?